internal/services/migrate_rewards.go

```diff
@@ -0,0 +1,107 @@
+package services
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	pb_defs "github.com/DIMO-Network/device-definitions-api/pkg/grpc"
+	"github.com/DIMO-Network/rewards-api/models"
+	"github.com/DIMO-Network/shared/db"
+	"github.com/ericlagergren/decimal"
+	"github.com/rs/zerolog"
+	"github.com/volatiletech/sqlboiler/v4/types"
+)
+
+func MigrateRewardsController(ctx context.Context, logger *zerolog.Logger, pdb db.Store, allIntegrations *pb_defs.GetIntegrationResponse, week int) error {
+
+	integrsByID := make(map[string]*pb_defs.Integration)
+	for _, integr := range allIntegrations.Integrations {
+		integrsByID[integr.Id] = integr
+	}
+
+	rewards, err := models.Rewards(
+		models.RewardWhere.IssuanceWeekID.LT(week),
+		models.RewardWhere.Tokens.GT(types.NewNullDecimal(decimal.New(0, 0))),
+	).All(ctx, pdb.DBS().Reader)
+	if err != nil {
+		return errors.New("failed to fetch rewards")
+	}
+
+	for _, reward := range rewards {
+		logger.Info().Int("Issuance Week", reward.IssuanceWeekID).Str("DeviceID", reward.UserDeviceID).Msg("Starting reward token migration")
+
+		if len(reward.IntegrationIds) == 0 {
+			continue
+		}
+
+		var sd *pb_defs.Integration
+		var ad *pb_defs.Integration
+
+		for _, id := range reward.IntegrationIds {
+			integration := integrsByID[id]
+
+			if integration.ManufacturerTokenId == 0 { // Synthetic
+				sd = integration
+			} else {
+				ad = integration
+			}
+		}
+
+		if ad == nil {
+			ad = &pb_defs.Integration{
+				Points: 0,
+			}
+		}
+
+		if sd == nil {
+			sd = &pb_defs.Integration{
+				Points: 0,
+			}
+		}
+
+		qry := `UPDATE
+			rewards_api.rewards
+			SET
+				synthetic_device_points = $1,
+				aftermarket_device_points = $2,
+			synthetic_device_tokens = coalesce(div(
+				$1 * tokens::INTEGER,
+				(SELECT
+					sum(streak_points + integration_points)
+					FROM rewards_api.rewards
+					WHERE
+						issuance_week_id = $3
+						AND user_device_id = $4
+				)
```

Comment:
Could we replace this whole subquery by `streak_points + integration_points`? I'm not sure what this is adding.